test/test_ops.py

```diff
@@ -1107,114 +1107,149 @@ def test_bbox_convert_jit(self):
+        assert_close(actual_box, expected_box, tolerance)
 
-        # Check for float boxes
-        for dtype in [torch.float16, torch.float32, torch.float64]:
-            box_tensor = torch.tensor(
-                [
-                    [285.3538, 185.5758, 1193.5110, 851.4551],
-                    [285.1472, 188.7374, 1192.4984, 851.0669],
-                    [279.2440, 197.9812, 1189.4746, 849.2019],
-                ],
-                dtype=dtype,
-            )
-            expected = torch.tensor([[1.0, 0.9933, 0.9673], [0.9933, 1.0, 0.9737], [0.9673, 0.9737, 1.0]])
-            iou_check(box_tensor, expected, tolerance=0.002 if dtype == torch.float16 else 1e-4)
-
-    def test_iou_jit(self):
-        box_tensor = torch.tensor([[0, 0, 100, 100], [0, 0, 50, 50], [200, 200, 300, 300]], dtype=torch.float)
+    def _run_jit_test(self, test_input):
+        box_tensor = torch.tensor(test_input, dtype=torch.float)
         TOLERANCE = 1e-3
-        expected = ops.box_iou(box_tensor, box_tensor)
-        scripted_fn = torch.jit.script(ops.box_iou)
-        scripted_iou = scripted_fn(box_tensor, box_tensor)
-        torch.testing.assert_close(scripted_iou, expected, rtol=0.0, atol=TOLERANCE)
+        expected = self._perform_box_operation(box_tensor)
+        scripted_area = self._perform_box_operation(box_tensor)
+        torch.testing.assert_close(scripted_area, expected, rtol=0.0, atol=TOLERANCE)
 
 
-class TestGenBoxIou:
-    def test_gen_iou(self):
-        def gen_iou_check(box, expected, tolerance=1e-4):
-            out = ops.generalized_box_iou(box, box)
-            torch.testing.assert_close(out, expected, rtol=0.0, check_dtype=False, atol=tolerance)
+class TestBoxArea(BoxTestBase):
+    __DEFAULT_TOLERANCE = 1e-4
+    __TOLERANCE_0_DOT_05 = 0.05
+    __INT_TEST_INPUT = [[0, 0, 100, 100], [0, 0, 0, 0]]
+    __INT_EXPECTED_OUTPUT = [10000, 0]
+    __FLOAT_32_N_64_TEST_INPUT = [
+        [285.3538, 185.5758, 1193.5110, 851.4551],
+        [285.1472, 188.7374, 1192.4984, 851.0669],
+        [279.2440, 197.9812, 1189.4746, 849.2019],
+    ]
+    __FLOAT_32_N_64_EXPECTED_OUTPUT = [604723.0806, 600965.4666, 592761.0085]
+    __FLOAT_16_TEST_INPUT = [
+        [285.25, 185.625, 1194.0, 851.5],
+        [285.25, 188.75, 1192.0, 851.0],
+        [279.25, 198.0, 1189.0, 849.0],
+    ]
+    __FLOAT_16_EXPECTED_OUTPUT = [605113.875, 600495.1875, 592247.25]
+
+    def _target_fn(self) -> Tuple[bool, callable]:
+        return (False, ops.box_area)
+
+    @pytest.mark.parametrize(
+        "test_input, dtype, tolerance, expected",
+        [
+            pytest.param(__INT_TEST_INPUT, torch.int8, __DEFAULT_TOLERANCE, __INT_EXPECTED_OUTPUT),
+            pytest.param(__INT_TEST_INPUT, torch.int16, __DEFAULT_TOLERANCE, __INT_EXPECTED_OUTPUT),
+            pytest.param(__INT_TEST_INPUT, torch.int32, __DEFAULT_TOLERANCE, __INT_EXPECTED_OUTPUT),
+            pytest.param(__INT_TEST_INPUT, torch.int64, __DEFAULT_TOLERANCE, __INT_EXPECTED_OUTPUT),
```

Comment:
I am a bit confused on this comment. If what you are suggesting here is to simply replace line 1168 to 1171 to be ```pytest.param(__INT_TEST_INPUT, [torch.int8, torch.int16, ...], __DEFAULT_TOLERANCE, __INT_EXPECTED_OUTPUT)``` then this result in single test run with dtypes array as parameter. This doesn't matches goal here as I then need to run loop in single test run to verify 4 different types, which is should be better parameterised.

If what you are suggesting here is to have dtypes parameterised seperated like ```@pytest.mark.parametrize("dtype", pytest.param([torch.int8, torch.int16, ...])``` then I found this runs as cartesian product of all possible combination of parametrised parameters, which includes incorrect combination as float input with int dtypes. The only way to avoid this is to have define separate method for int and float, which I didn't choose because it will create some level of duplication.

But I do see your concern here with duplicated params. I will see what I can do here.

Reply:
Sorry I should have been clearer. I was implying that you will replace the `dtype` with `dtypes` (plural) and refactor the test to loop for each provided type within the list. This is an optional nit that we can leave to decide at the very end. :)